app/services/PaymentAPIService.scala

```diff
@@ -64,25 +64,56 @@ class PaymentAPIService(wsClient: WSClient, paymentAPIUrl: String) {
       .execute()
   }
 
-  def isDuplicatePaymentResponse(response: WSResponse): Boolean = {
+  def decodePaymentResponse(response: WSResponse): Option[PaymentApiError] = {
     decode[PaymentApiError](response.body).fold(
       failure => {
         val failureMessage = SafeLogger.LogMessage(
           s"Unable to decode PaymentAPIError: ${response.body}. Message is ${failure.getMessage}",
           "Unable to decode PaymentAPIError. See logs for details"
         )
         SafeLogger.error(failureMessage)
-        false
+        None
       },
-      paymentApiError => paymentApiError.error.errorName.contains("PAYMENT_ALREADY_DONE")
+      resp => {
+        Some(resp)
+      }
     )
   }
 
+  def logErrorResponse(paymentApiError: Option[PaymentApiError]): Unit = {
+    paymentApiError.foreach(err => {
+      if (err.error.errorName.contains("INSTRUMENT_DECLINED")) {
```

Comment:
I would just log the whole error regardless of the error name (I don't think we need a conditional here)

Reply:
As discussed, I'm keen to keep "instrument_declined" errors out of our Sentry tool for two reasons:
1) This error is not something we care about - it means Paypal didn't like the users card details.
2) This error is by far the most commonly occurring error. Its presence will prevent us from detecting the errors we actually do need to take action on.